src/peripherals/stm_flash.c

```diff
@@ -117,6 +121,26 @@ u8 stm_flash_program(u32 address, u8 data[], u8 length)
   return FLASH_OK;
 }
 
+/** Callback to read STM32F4's hardcoded unique ID.
+ * Sends STM32F4 unique ID (12 bytes) back to host.
+ */
+void stm_unique_id_callback(u16 sender_id, u8 len, u8 msg[], void* context)
```

Comment:
Actually, I guess it's ROM, not flash. So yeah it probably makes sense to put somewhere else.